### PR TITLE
supports source type and abstract out findComponents util

### PR DESCRIPTION
**Pull request by BRKalow**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -4,15 +4,16 @@ import matter from 'gray-matter'
...
+import { findComponent } from './utils/find-component'
 import components from './__swingset_components'
 
 export function createStaticPaths() {
   return function getStaticPaths() {
     return {
       paths: Object.values(components).map((componentConfig) => {
-        return { params: { swingset: [componentConfig.slug] } }
+        return { params: { swingset: ['components', componentConfig.slug] } }
       }),
```
Comment (BRKalow) [inline server.js:15]: when we work in adding support for the standalone docs pages, we'll just have to make sure we prepend `'docs'` to the param array

File: server.js [hunk 2]
```diff
@@ -23,49 +24,55 @@ export function createStaticProps(swingsetOptions = {}) {
...
     }))
 
-    // get the full source and metadata for the component that's rendered on the current page
-    const currentComponentData = Object.values(components).find(
-      (componentConfig) => {
-        return componentConfig.slug === params.swingset[0]
-      }
-    )
+    // the first route segment dictates how we load data
+    const sourceType = params.swingset[0]
```
Comment (BRKalow) [inline server.js:31]: Open to naming suggestions here, but `sourceType` seemed reasonable

File: utils/get-peer-components.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export function getPeerComponents(component, components) {
```
Comment (BRKalow) [inline utils/get-peer-components.js:1]: Saw a lil cleanup opportunity here and took it.

File: page.jsx [hunk 4]
```diff
@@ -6,10 +6,12 @@ import Link from 'next/link'
 import { MDXRemote } from 'next-mdx-remote'
 import createScope from './utils/create-scope'
 import { useRestoreUrlState, setUrlState } from './utils/url-state'
+import { findComponent } from './utils/find-component'
 import components from './__swingset_components'
+import { getPeerComponents } from './utils/get-peer-components'
 
 export default function createPage(swingsetOptions = {}) {
-  return function Page({ mdxSource, navData }) {
+  return function Page({ mdxSource, navData, initialParams }) {
```
Comment (BRKalow) [inline page.jsx:14]:
```suggestion
  return function Page({ mdxSource, navData }) {
```

**Review by jescalan — APPROVED** [review]

**Pull request merged**